src/keys/keypair.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+//! Module providing keys, keypairs, and signatures.
+//!
+//! The easiest way to get a `PublicKey` is to create a random `Keypair` first through one of the
+//! `new` functions. A `PublicKey` can't be generated by itself; it must always be derived from a
+//! secret key.
+
+use crate::{Error, Result};
+use crate::{PublicKey, SecretKey, Signature, SignatureShare};
+
+use ed25519_dalek::Signer;
+use rand::{CryptoRng, Rng};
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Debug, Formatter};
+use threshold_crypto::{self, serde_impl::SerdeSecret};
+use unwrap::unwrap;
+
+/// Wrapper for different keypair types.
+#[derive(Serialize, Deserialize)]
+pub enum Keypair {
+    /// Ed25519 keypair.
+    Ed25519(ed25519_dalek::Keypair),
+    /// BLS keypair.
+    Bls(BlsKeypair),
+    /// BLS keypair share.
+    BlsShare(BlsKeypairShare),
+}
+
+// Need to manually implement this due to a missing impl in `Ed25519::Keypair`.
+impl Clone for Keypair {
+    fn clone(&self) -> Self {
+        match self {
+            Self::Ed25519(keypair) => Self::Ed25519(unwrap!(ed25519_dalek::Keypair::from_bytes(
```

Comment:
This is dangerous, both Clone and unwrap!

Reply:
So I've taken the olde impl we already had here.

I know we don't want `unwrap` ideally... but in a `clone` or `from` that can't have a `Result` type what's the best way?

Is it actually not to `clone`, but some custom func? (I guess just not use the trait?)